### PR TITLE
SymIntify functorch vmap

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #101413
* __->__ #101409

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

File: aten/src/ATen/functorch/DynamicLayer.cpp [hunk 1]
```diff
@@ -28,7 +28,7 @@ void setDynamicLayerFrontBackKeysIncluded(bool included) {
 DynamicLayer::DynamicLayer(
     TransformType transform_type,
     int64_t layerId,
-    optional<int64_t> batchSize,
+    optional<c10::SymInt> batchSize,
```
Comment (Skylion007) [inline aten/src/ATen/functorch/DynamicLayer.cpp:31]: You probably are missing a few std::move in these files too

**Review by Skylion007 — APPROVED** [review]
A lot of missing std::moves in the call dispatch of this PR, please add them.

**Review by Skylion007 — APPROVED** [review]

**Review by zou3519 — APPROVED** [review]

**Pull request closed without merging**